matsim/src/main/java/org/matsim/core/population/PopulationUtils.java

```diff
@@ -1093,4 +1080,49 @@ public static boolean comparePopulations( Population population1, Population pop
 		boolean result = PopulationUtils.equalPopulation( population1, population2 );
 		return result ;
 	}
+
+	// ---
+
+	public static Object getPersonAttribute( HasPlansAndId person, String key, Population population ) {
```

Comment:
Why HasPlanAndId, when all other methods accept a `Person`? Shouldn't then all other methods accept a `HasPlansAndId` as well?

Reply:
There is actually a comment in `HasPlansAndId`:
https://github.com/matsim-org/matsim/blob/7ca3b868329c137060e8a0d245b4c2a7f451354b/matsim/src/main/java/org/matsim/api/core/v01/population/HasPlansAndId.java#L8-L14
The issue is that the main reason for the existence of `HasPlansAndId` is the strategy manager, and the strategy manager is also used by the carriers (and should also be used by some other things, such as paratransit).  And so it is _not_ only a person that is passed into it.

Now evidently we could decide that the subpop functionality is only there for the persons, but I don't see why.